### DISCUSSION TITLE
Add at the Racks the Cable Types too

### DISCUSSION THREAD

**Discussion by Dave6998** [netbox-community/netbox#6928] [Ideas]
### Change Type

Addition

### Area

Functionality/features

### Proposed Changes

Hi, 

it would help if we had the option to choose a cable/socket at the rack.
Much better it would be to choose also the plug or socket.
I would prefer to manage it by my self on the user level like make the cable/plug or socket types what i can choose finally on connecting.
Not only a interface, because we have here also just sockets or plugs in the racks.
For example: 
Cable type: Single Mode 
Plug: SC-PC

BR, Dave

**Comment by candlerb** [comment]
In Netbox, Cables don't have types for their connectors: however the interfaces, front ports, rear ports, console ports, port ports (etc) all do.  These are not user-definable, but if there are some specific types you need which don't currently exist, you can raise a feature request.  This is because the Netbox team want to maintain control and consistency over the types.  See e.g. issue #84 and related ones for further discussion.

The cable endpoints are implied by what they connect to: e.g. if you create a cable between a 1000baseTX and an 8P8C, then it's implicitly an RJ45-RJ45 cable.  You can select the cable *type* (e.g. CAT5e, CAT6 etc), again from another fixed list.

There are certainly several limitations in the data model as it relates to cabling.  These come immediately to mind:
- there is no modelling of interface modules, like SFPs.  So a cable "connects" directly into an interface in Netbox, rather than connecting into an SFP (and the SFP plugging into the interface)
- there is no modelling of individual fibre strands, i.e. that many SFPs require two fibres.  So a "cable" might represent a fibre pair, if connected to an interface which has a dual LC connector.  This doesn't work well when your fibre plant carries a mixture of duplex connections and single-fibre (BiDi/PON)
- there is no modelling of multiple interfaces on the same physical port (e.g. QSFP configured as 4x10G interfaces)
- there is no modelling of breakout cables (e.g. QSFP to 4xSFP). All cables in Netbox have exactly two ends.
- you can't have cables connected at one end only, or cables which connect to other cables

**Comment by Dave6998** [comment]
Thank you for the answer.
I understand what you mean.
I know it from my experience, its for many ppl important to have the option end a cable in a rack with a socket or plug.
Aslo have just a cable with plugs or sockets betweet 2 racks.
